enhancements/monitoring/restart-metrics.md

```diff
@@ -0,0 +1,122 @@
+---
+title: restart-metrics
+authors:
+  - "@rphillips"
+reviewers:
+  - "@smarterclayton"
+  - "@sjenning"
+  - "@mrunalp"
+approvers:
+  - "@brancz"
+  - "@bparees"
+  - "@squat"
+  - "@s-urbaniak"
+  - "@metalmatze"
+  - "@paulfantom"
+  - "@LiliC"
+  - "@pgier"
+  - "@simonpasquier"
+creation-date: 2020-03-23
+last-updated: 2020-03-23
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Service Restart Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+There is not any insight with systemd units restarting. Metrics of this sort are
+important to know if Kubelet, crio, or other system services are crashing and
+restarting. Restart metrics are going to be extremely important since the
+Kubelet is going to be changed to exit on a crash, thus restarting. Previous
+behavior of the Kubelet is to recover() from the crash and not exit.
+
+## Motivation
+
+Systemd unit restart metrics of a unit are vitally important for system
+administrators to see the health of the Kubelet, crio, and other vitally important
+system services.
+
+### Goals
+
+- Systemd unit restart metrics need to be propogated through to monitoring, and
+  the alerting system to alert system administrators to problems within the
+  system.
+
+### Non-Goals
+
+- Defining alerts is not in the scope of this proposal. Alerts can be defined
+  once we see some sample cluster data.
+
+### Current Behavior
+
+node_exporter is running in a non-privileged container - as user `nobody` - and
+cannot connect to the DBUS socket to communicate with systemd. Monitoring has
+concerns the underlying [systemd
+collector](https://github.com/prometheus/node_exporter/blob/master/collector/systemd_linux.go)
+is not performant enough.
+
+## Proposal
+
+- This proposal is to add a privileged sidecar to node_exporter running
+  [systemd_exporter](https://github.com/povilasv/systemd_exporter])
+
+- systemd_exporter will be configured to write metrics to node_exporters
```

Comment:
This includes deploying the systemd_exporter, we use jsonnet for all generation of our asset manifests and this should be done the same way for systemd_exporter. This note is more of a warning for any estimates or FYI.

Reply:
I think using a daemonset to deploy this exporter might make sense here? Also this then needs to be forked into openshift org, I assume you would take ownership of that?

Reply:
We should be able to use the current daemonset for node_exporter and add a sidecar container definition. I believe we are using the upstream version of node_exporter, and should be able to leverage the upstream version of systemd_exporter without needing a fork.

Please correct me if I'm wrong on this.

Reply:
96cf786 contains a blurb on where the extra container would go.

Reply:
We use upstream in the sense we do not modify it, but via the fork yes like all our other components. 

> Update 2: added a blurb that Node team will own the image, but the configuration and deployment will be done by
sgtm 👍